core/common/src/main/java/alluxio/underfs/UnderFileSystemConfiguration.java

```diff
@@ -23,24 +23,25 @@
  * {@link RuntimeException} if the key is not found in both configurations..
  */
 public final class UnderFileSystemConfiguration {
-  private final boolean mReadOnly;
-  private final boolean mShared;
-  private final Map<String, String> mUfsConf;
+  private boolean mReadOnly;
+  private boolean mShared;
+  private Map<String, String> mUfsConf;
 
-  private static final UnderFileSystemConfiguration DEFAULTS = new UnderFileSystemConfiguration();
+  private static final UnderFileSystemConfiguration EMPTY_CONFIG =
+      new UnderFileSystemConfiguration();
 
   /**
-   * Constructs a new instance of the configuration for a UFS.
-   *
-   * @param readOnly whether only read operations are permitted to UFS
-   * @param shared whether the mount point is shared with all Alluxio users
-   * @param ufsConf the user-specified UFS configuration as a map
+   * @return default UFS configuration
    */
-  public UnderFileSystemConfiguration(boolean readOnly, boolean shared,
-      Map<String, String> ufsConf) {
-    mReadOnly = readOnly;
-    mShared = shared;
-    mUfsConf = ufsConf;
+  public static UnderFileSystemConfiguration defaults() {
+    return new UnderFileSystemConfiguration();
+  }
+
+  /**
+   * @return empty UFS configuration
+   */
+  public static UnderFileSystemConfiguration empty() {
```

Comment:
I would prefer to just stick with `defaults`. We should not share instances of `UnderFileSystemConfiguration` since the overhead of creating the object should be negligible and it gives us isolation.